cli/src/cli_util.rs

```diff
@@ -1012,7 +1012,19 @@ impl WorkspaceCommandHelper {
         let immutable = revset_util::parse_immutable_expression(&self.revset_parse_context())?;
         let mut expression = self.attach_revset_evaluator(immutable)?;
         expression.intersect_with(&to_rewrite_revset);
-        if let Some(commit_id) = expression.evaluate_to_commit_ids()?.next() {
+
+        let mut commit_id_iter = match expression.evaluate_to_commit_ids() {
+            Err(UserRevsetEvaluationError::Resolution(RevsetResolutionError::NoSuchRevision {
+                name,
+                candidates: _,
+            })) => Err(user_error_with_hint(
+                format!(r#"The revision "{name}" used in immutable_heads() does not exist"#),
+                r#"If you configured specific branch names consider wrapping them with "present()" or using the functions "branches()" / "remote_branches()"."#,
+            )),
+            other => other.map_err(Into::into),
+        }?;
```

Comment:
nit: perhaps, `config_error_with_message().hinted()` is more appropriate here.

And I personally think we don't need to explicitly check the error type. `NoSuchRevision` isn't the only error that the user could make.

Reply:
Good point.

It would also be good to have a test or two in https://github.com/martinvonz/jj/blob/main/cli/tests/test_immutable_commits.rs

Reply:
hmm, the hint in this case does depend on the specific type of error and to me it seems like `RevsetEvaluationError` and `RevsetResolutionError::StoreError` would not be config errors. 
It'd make sense for the other ones, but in that case I'd either drop the hint or only have that hint in case of that specific error.

Reply:
> it seems like `RevsetEvaluationError` and `RevsetResolutionError::StoreError` would not be config errors.

It's technically correct that `StoreError` isn't a config error, but I think it's less likely to occur, and the user can see what happened if we provided the original error.

Regarding the hint, I prefer removing it. If the user configured a specific branch name with a specific remote, the resolution error shouldn't be suppressed by `present()`.

Reply:
Updated, will add some tests later

Reply:
Added some tests